packages/muelu/src/Misc/MueLu_AggregateQualityEstimateFactory_def.hpp

```diff
@@ -460,6 +473,70 @@ namespace MueLu {
     }
   }
 
+
+
+template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  void AggregateQualityEstimateFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::ComputeAggregateSizes(RCP<const Matrix> A, RCP<const Aggregates> aggs, RCP<LocalOrdinalVector> agg_sizes) const {
+
+    ArrayRCP<LO> aggSortedVertices, aggsToIndices, aggSizes;
+    ConvertAggregatesData(aggs, aggSortedVertices, aggsToIndices, aggSizes);
+
+    // Iterate over each node in the aggregate
+    auto data = agg_sizes->getDataNonConst(0);
+    for (LO i=0; i<(LO)aggSizes.size(); i++)
+      data[i] = aggSizes[i];      
+}
+
+
+
+template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
+  void AggregateQualityEstimateFactory<Scalar, LocalOrdinal, GlobalOrdinal, Node>::OutputAggSizes(const Level& level, RCP<const LocalOrdinalVector> agg_sizes) const {
+
+    ParameterList pL = GetParameterList();
+    using MT = magnitudeType;
+
+    ArrayRCP<const LO> data = agg_sizes->getData(0);
+
+
+    if (pL.get<bool>("aggregate qualities: file output")) {
+      std::string filename = pL.get<std::string>("aggregate qualities: file base")+".sizes."+std::to_string(level.GetLevelID());
+      Xpetra::IO<LO,LO,GO,Node>::Write(filename, *agg_sizes);
+    }
+
+    {
+      size_t n = (size_t)agg_sizes->getLocalLength();
+
+      std::vector<MT> tmp;
+      tmp.reserve(n);
+
+      for (size_t i=0; i<n; ++i) {
+        tmp.push_back(Teuchos::as<MT>(data[i]));
+      }
+
+      std::sort(tmp.begin(), tmp.end());
+
+      Teuchos::ArrayView<const double> percents = pL.get<Teuchos::Array<double> >("aggregate qualities: percentiles")();
+
+      GetOStream(Statistics1) << "AGG SIZE    HEADER     : | LEVEL |  TOTAL  |";
+      for (auto percent : percents) {
+        GetOStream(Statistics1) << std::fixed << std::setprecision(4) <<100.0*percent << "% |";
+      }
+      GetOStream(Statistics1) << std::endl;
+
+      GetOStream(Statistics1) << "AGG SIZE    PERCENTILES: | " << level.GetLevelID() << " | " << n << "|";
+      for (auto percent : percents) {
+        size_t i = size_t(n*percent);
+        i = i < n ? i : n-1u;
+        i = i > 0u ? i : 0u;
```

Comment:
These would be easier (for me) to parse if they were written `i = (i <n) ? i : n-1u;` and `i = (i > 0u) ? i : 0u;`.